### PR TITLE
cloudsql-postgresql e2e

### PR DESCRIPTION
**Pull request by bharatgulati**
CloudSQL-PostgreSQL e2e

### CODE REVIEW

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  }
+
+  public static void deleteTable(String table, String schema) throws SQLException, ClassNotFoundException {
+    try (Connection connection = getCloudSqlConnection();
+         Statement statement = connection.createStatement()) {
+        String dropTableQuery = "Drop Table " + schema + "." + table;
+        statement.executeUpdate(dropTableQuery);
+      }
+  }
+//  public static void deleteTable(String table)
```
Comment (sumitjnn) [inline cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java]: Remove this

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      String datatypesColumnsList = PluginPropertyUtils.pluginProp("datatypesColumnsList");
+      statement.executeUpdate("INSERT INTO " + schema + "." + sourceTable + " " + datatypesColumnsList + " " +
+                                datatypesValues);
+    }
+  }
+
+  public static void createTargetTable(String targetTable, String schema) throws SQLException, ClassNotFoundException {
+    try (Connection connection = getCloudSqlConnection();
+         Statement statement = connection.createStatement()) {
+      String datatypesColumns = PluginPropertyUtils.pluginProp("datatypesColumns");
```
Comment (sumitjnn) [inline cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java]: Use constants

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
```
Comment (sumitjnn) [inline cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java:52]: Most of the code in this class seems to be same as the one in MySQL test cloudsql-mysql-plugin/src/e2e-test/java/io/cdap/plugin/cloudmysql/BQValidation.java, are there any differences except how connection is created `CloudSqlPostgreSqlClient.getCloudSqlConnection()` ?

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                                               PluginPropertyUtils.pluginProp("schema"));
+    BeforeActions.scenario.write("CLOUDPOSTGRESQL Target table - " + PluginPropertyUtils.pluginProp("targetTable")
+                                   + " created successfully");
+  }
+
+  @After(order = 2, value = "@CLOUDSQLPOSTGRESQL_TARGET_TEST")
+  public static void dropTargetTable() throws SQLException, ClassNotFoundException {
+    CloudSqlPostgreSqlClient.deleteTable(PluginPropertyUtils.pluginProp("targetTable"),
+                                         PluginPropertyUtils.pluginProp("schema"));
+    BeforeActions.scenario.write("CLOUDMYSQL Target Table - " + PluginPropertyUtils.pluginProp("targetTable") +
```
Comment (sumitjnn) [inline cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java]: CLOUDPOSTGRESQL

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                                               PluginPropertyUtils.pluginProp("schema"));
+    BeforeActions.scenario.write("CLOUDPOSTGRESQL Source table - " + PluginPropertyUtils.pluginProp("sourceTable")
+                                   + " created successfully");
+  }
+
+  @After(order = 2, value = "@CLOUDSQLPOSTGRESQL_SOURCE_TEST")
+  public static void dropSourceTable() throws SQLException, ClassNotFoundException {
+    CloudSqlPostgreSqlClient.deleteTable(PluginPropertyUtils.pluginProp("sourceTable"),
+                                         PluginPropertyUtils.pluginProp("schema"));
+    BeforeActions.scenario.write("CLOUDMYSQL Source Table - " + PluginPropertyUtils.pluginProp("sourceTable") +
```
Comment (sumitjnn) [inline cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java]: CLOUDPOSTGRESQL

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java (hunk 1 above)
Comment (bharatgulati) [reply cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java]: Done.

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java (hunk 2 above)
Comment (bharatgulati) [reply cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/CloudSqlPostgreSqlClient.java]: Replaced with Constants.

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java (hunk 3 above)
Comment (bharatgulati) [reply cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java:52]: The difference is in the supported data types of  ```CloudSQLPostgreSQL``` mostly and how the transformations are done for each datatype for Data Validation.

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java (hunk 4 above)
Comment (bharatgulati) [reply cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java]: Done.

File: cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java (hunk 5 above)
Comment (bharatgulati) [reply cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetUpHooks.java]: Done.

**Review by sumitjnn — APPROVED** [review]

**Review by yjhawar — APPROVED** [review]

**Comment by yjhawar** [comment]
Why are builds failing? Is the fix for build failures in progress?

**Comment by bharatgulati** [comment]
> Why are builds failing? Is the fix for build failures in progress?

The Build was failing and now it is fixed. We are waiting for an approval on the CloudSQL-mySQL PR build failures changes.(https://github.com/data-integrations/database-plugins)

**Comment by bharatgulati** [comment]
A few tests are failing due to an intermittent chrome version issue, and we have reported a bug for this. (https://cdap.atlassian.net/browse/PLUGIN-1649)

**Comment by Vipinofficial11** [comment]
Merging the PR with intermittent failure as discussed in the calls.

**Pull request merged**